util/tracing/goexectrace: TestSimpleFlightRecorder failed

util/tracing/goexectrace.TestSimpleFlightRecorder [failed](https://mesolite.cluster.engflow.com/invocations/default/16c84881-d8ac-4dc3-8a38-25489d335296?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdXRpbC90cmFjaW5nL2dvZXhlY3RyYWNlOmdvZXhlY3RyYWNlX3Rlc3Q=) on master @ [7b66342542aabc48e76591002acec879733811d2](https://github.com/cockroachdb/cockroach/commits/7b66342542aabc48e76591002acec879733811d2):


```
=== RUN   TestSimpleFlightRecorder
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestSimpleFlightRecorder3669911446
    test_log_scope.go:76: use -show-logs to present logs inline
    simple_flight_recorder_test.go:87: -- test log scope end --
test logs left over in: outputs.zip/logTestSimpleFlightRecorder3669911446
--- FAIL: TestSimpleFlightRecorder (3.68s)
=== RUN   TestSimpleFlightRecorder/writes_a_file_when_enabled
    simple_flight_recorder_test.go:60: 
        	Error Trace:	pkg/util/tracing/goexectrace/simple_flight_recorder_test.go:60
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:236
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/util/tracing/goexectrace/simple_flight_recorder_test.go:54
        	Error:      	"0" is not greater than "0"
        	Test:       	TestSimpleFlightRecorder/writes_a_file_when_enabled
--- FAIL: TestSimpleFlightRecorder/writes_a_file_when_enabled (1.37s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSimpleFlightRecorder.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

util/tracing/goexectrace.TestSimpleFlightRecorder [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20092355?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20092355?buildTab=artifacts#/) on master @ [7b66342542aabc48e76591002acec879733811d2](https://github.com/cockroachdb/cockroach/commits/7b66342542aabc48e76591002acec879733811d2):


```
=== RUN   TestSimpleFlightRecorder
metamorphic: use COCKROACH_RANDOM_SEED=-6407238225338096202 for reproduction
initialized metamorphic constant "parse-json-impl" with value 1
initialized metamorphic constant "coldata-batch-size" with value 1610
initialized metamorphic constant "span-reuse-rate" with value 97
initialized metamorphic constant "COCKROACH_ENABLE_HDR_HISTOGRAMS" with value true
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/e80c91308ef01ef2193e4a1534570d85/logTestSimpleFlightRecorder470462911
    test_log_scope.go:76: use -show-logs to present logs inline
    simple_flight_recorder_test.go:87: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e80c91308ef01ef2193e4a1534570d85/logTestSimpleFlightRecorder470462911
--- FAIL: TestSimpleFlightRecorder (3.17s)
=== RUN   TestSimpleFlightRecorder/writes_a_file_when_enabled
    simple_flight_recorder_test.go:60: 
        	Error Trace:	pkg/util/tracing/goexectrace/simple_flight_recorder_test.go:60
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:236
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/util/tracing/goexectrace/simple_flight_recorder_test.go:54
        	Error:      	"0" is not greater than "0"
        	Test:       	TestSimpleFlightRecorder/writes_a_file_when_enabled
--- FAIL: TestSimpleFlightRecorder/writes_a_file_when_enabled (1.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSimpleFlightRecorder.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/util/tracing/goexectrace/simple_flight_recorder_test.go
+++ b/pkg/util/tracing/goexectrace/simple_flight_recorder_test.go
@@ -56,12 +56,22 @@ func TestSimpleFlightRecorder(t *testing.T) {
 				return errors.New("flight recorder is not enabled")
 			}
 			files, err := os.ReadDir(dir)
-			require.NoError(t, err)
-			require.Greater(t, len(files), 0)
+			if err != nil {
+				return err
+			}
+			if len(files) == 0 {
+				return errors.New("no files written")
+			}
 			fi, err := os.Stat(filepath.Join(dir, files[0].Name()))
-			require.NoError(t, err)
-			require.Greater(t, fi.Size(), int64(0))
-			require.Regexp(t, fileMatchRegexp, files[0].Name())
+			if err != nil {
+				return err
+			}
+			if fi.Size() == 0 {
+				return errors.New("file is empty")
+			}
+			if !fileMatchRegexp.MatchString(files[0].Name()) {
+				return errors.New("file name does not match expected pattern")
+			}
 			return nil
 		})
 	})

